### ISSUE TITLE
Duplicates - episodes getting scrobbled 2-3 times

### ISSUE THREAD

**Issue by insurw** [Razzeee/script.trakt#169]
Episodes are very often getting scrobbled 2-3 times all with the exact same date and time.
XBMC log: https://drive.google.com/file/d/0BxIAJYOpIauFTHVnN1c4WWtJZVk/view?usp=sharing

In my case it's because of using the **same trakt account on multiple XBMC profiles**. A solution for now is to automatically restart XBMC after switching profiles (e.g. using [EventGhost](http://www.eventghost.org/)). Though if you're using the addon "XBMC Library Auto Update" change its startup delay setting to 1+ min. Because if you've trakt's setting "Sync collection on library update or cleaning" enabled you'll get duplicates.

**Comment by t-mathis** [comment]
I apologize if this is the incorrect place to mention this, but the same issue is happening for TV episodes as well. I have 3 shows that fail to sync due to 500 Internal Server Errors. They have 445, 772, and 548 episodes respectively. I'm not sure what the threshold is, but I had a show with 269 episodes sync fine. I also have a show with 361 episodes, but I cannot confirm if that synced, since I have it set up differently.

(The changed files did allow my movies to be synced though :) )

**Comment by nate1280** [comment]
I'll have to look into that, splitting a TV show will be a bit more complicated.  No idea why this is cropping up now, had no issues before.

If I figure out a fix, it will be in a separate merge request.

**Comment by rectifyer** [comment]
Weird about the 500 server errors, I'll have to look into that.

**Comment by Razzeee** [comment]
Well your log is basically saying your not scrobbling, every playback has:

    script.trakt] service: [traktPlayer] onPlayBackStarted() - Non-library file, not enough data for scrobbling, skipping.

**Comment by insurw** [comment]
The addon IS scrobbling movies and shows. E.g. Once Upon a Time that was scrobbled 3 times, not sure where it happens in the log but it should be there.

All occurrences of "not enough data for scrobbling" is after I've started playing a video from the Youtube or SVT Play addons.

**Comment by RavRob** [comment]
I have the exact same thing happening.  I thought it might have been caused by the addon but could never pin point it.  Now that I see others have this issue, I will enable my log and hope it happens with the log going.  It is not happening very often but when it does, it is normally 3 times.  

I have only one profile. 
Using Genesis addon
Windows 8.1 
Isenguard 15 Alpha 1

**Comment by insurw** [comment]
Here's another example.
"justified.s06e10" was added twice to trakt.tv

https://drive.google.com/file/d/0B1-kqUA_uxWGZVlydnl5RjVPTjg/view?usp=sharing

**Comment by Razzeee** [comment]
Just to be safe, are you guys using any other apps that are accessing your trakt profile? Series Guide, the Plex app etc

**Comment by Razzeee** [comment]
Your log is really really interesting, for some reason everything gets called two times.

You manage to start another instance of the script here:
```
09:13:43 T:3568   DEBUG: [script.trakt] __main__: Loading 'script.trakt' version '3.0.0'
```
So everything we send gets interpreted twice. I would think this might be profile related.

**Comment by s7eele** [comment]
just information i thought might help, i went through my trakt profile thoroughly today to make sure i havent been affected by subject issue and its verified, this has not happened to me, profile sounds like a likely cause (especially if they use same trakt account) as @Razzeee mentioned but then again @RavRob is apparently seeing it with only one profile, hopefully u can catch a log of it, just for clarification:

not affected
android 4.4.2 (ouya)
kodi 14.2rc
genesis/salts/1ch_primewire

**Comment by insurw** [comment]
Ok. I hope it can get fixed. Going back to one XBMC profile again isn't an option, it's so messy mixing unseen and old stuff. Using different trakt accounts for different profiles (in my case different libraries) would be an option but it'd be great if you could use the same account. At least when you don't have the same libraries in the profiles.

**Comment by RavRob** [comment]
For me it happens so seldom that is very difficult to catch a good log.  Hopefully I will get one next time it happens.

**Comment by Razzeee** [comment]
@ravrob it's probably when a request gets retried. Will be really really hard to fix from my end of things. 
The server is sending an error to the client, but already has written some parts of the request. Then the client retries again, gets no error and we have a duplicate

**Comment by insurw** [comment]
> Will be really really hard to fix from my end of things. The server is sending an error to the client, but already has written some parts of the request. Then the client retries again, gets no error and we have a duplicate

So should I report it at support.trakt.tv? Or do you work with them? (i.e. forwarding this kind of bugs)

**Comment by Razzeee** [comment]
That's a seperate problem from yours.

**Comment by tknorris** [comment]
If everything is getting called twice, that suggests that there are two instances of the trakt script installed. That can happen when someone installed the script from github and installed it from a repository. You can double check in the addons directory to see if there are two script.trakt directories in it.

**Comment by insurw** [comment]
That's not the problem. I removed all traces of the old takt addon before switching to Razzeee's addon.

**Comment by dekes1** [comment]
I am experiencing the same issue.  I started new Trakt accounts just to test the add-on and everytime the script syncs it creates two entries in my movie collection.  Using version 3.0.0 of the script from the Kodi repo

Trakt account as example:   https://trakt.tv/users/dekesone/collection/movies/added

I can provide logs if it would be beneficial

**Comment by Razzeee** [comment]
Well the first time you change your profile, its shut down correctly. But it seems to fail after that.

**Comment by insurw** [comment]
It seems only scrobbling and manually starting the addon produces duplicates. If I set it to sync on library update it works fine.

**Comment by s7eele** [comment]
@insurw, im just grasping at straws based on the suggestions of others but figure it cant hurt, your response to the suggestion from @tknorris left me wondering if you may have misunderstood, did you look closely for a duplicate directory in the addons folder, for instance one might be named script.trakt while the other is named script.trakt-master or script.trakt-dev, this can happen when installing from github and it would indicate two copies of the script running as tk suggested, what you described sounded like you were just cleaning out the script.trakt directory, again im just grasping at straws so please dont take offense

**Comment by insurw** [comment]
@s7eele No, I deleted every trakt folder by searching for trakt in the XBMC folder (under Roaming).

**Comment by insurw** [comment]
I was going to post another log of the addon  working with Library updates as I mentioned before. But it seems I was wrong. If I stay on one profile it seems to work, but when I start switching back and forth the duplicates start to happen.

**Comment by tknorris** [comment]
I wonder if the service isn't getting shutdown properly when you exit a profile.

**Comment by tknorris** [comment]
That looks to be the problem based on this post: http://forum.kodi.tv/showthread.php?tid=92751&pid=882231#pid882231 According to that post, services started at login are not stopped at logoff as you would expect.

**Comment by Razzeee** [comment]
I'm not so sure about that, my logfile says that trakt shutsdown the first time I change the profile.
The othertimes I think kodi is forcing it to stop? Not sure as I have no logfile, but there is a message for scripts that need to long to shutdown beeing forced.

**Comment by tknorris** [comment]
If kodi was forcing the service to stop then you wouldn't expect multiple scrobbles...

**Comment by Razzeee** [comment]
And Martijn also reassures me that this is correct. See:

https://github.com/Razzeee/script.trakt/blob/master/addon.xml#L12
http://forum.kodi.tv/showthread.php?tid=92751&pid=1002362#pid1002362

**Comment by tknorris** [comment]
Yeah I saw martijn's post. The only thing I can think is maybe theres an xbmc bug causing the service not to be shutdown when it should be

**Comment by Razzeee** [comment]
Yep, thats what we need to test/reconstruct.

**Comment by insurw** [comment]
Well, I "solved" my problem by simply restarting XBMC after every profile switch. As I'm already using EventGhost for my remote and to switch profiles it was quite easy. I just added a delay, in my case 5 s, after the profile switch and then restart XBMC with cmd. I guess you could use the builtin restart function instead, but I already have a BAT file to taskkill & start XBMC.

**Comment by RavRob** [comment]
The new update 3.0.1 has started to scrobble everything twice.
My log
https://drive.google.com/open?id=0B13zfUny-eEWWm1RNy1xdTlrM1U&authuser=0

**Comment by tknorris** [comment]
I don't have access to your log, but I'm running 3.0.1 and not getting double scrobbles.

**Comment by s7eele** [comment]
same, no access to log but not seeing double scrobbles either,

**Comment by Razzeee** [comment]
@ravrob have you switch profiles since your last reboot?

**Comment by RavRob** [comment]
I do not use multiple profiles so I doubt that would be the same problem as the OP.  Sorry about the log not accessible.  I did that fairly late last night.  Pastebin will not accept such a large log.  I had to pause the TV shows about 10 min to the end so I could start my debug mode but it still created a very large log (over 3 megs) and pastebin only accepts up to 512 kb.

The weird thing is that it isn't all the time.  Once in a while it scrobble correctly but 99.9% of the time it scrobble double.  This is while using Genesis.  I will try with SALTS (the only other addon I use) and I will try also with local media see if the same still holds true.

I will try and post my log again tonight when I get home or maybe try and recreate a new, smaller log.

**Comment by RavRob** [comment]
OK here is my log.  This one was for a movie played in Genesis
http://www.xbmclogs.com/psbeeqeqz

**Comment by Razzeee** [comment]
Are you sure you've got the debug log activated? There is not one string in that log thats like "trakt"

**Comment by RavRob** [comment]
I just checked and the Trakt debug mode is on.   Not sure what might have
happened that it doesn't show in the log.

**Comment by Razzeee** [comment]
@RavRob have you also enabled the kodi debug logging?
https://github.com/Razzeee/script.trakt#creating-logfiles

**Comment by RavRob** [comment]
I enabled the Kodi debug towards the end of the show....when it got to
around 75% of the show, only to show the end when it would/should
scrobble.    I am going to do it again but this time from the beginning.  I
will post a new log again later.

**Comment by RavRob** [comment]
OK here is my new log.  Kodi debug mode enabled, rebooted Kodi and started the tv show.  I also made sure Trakt debug mode was enabled.
http://www.xbmclogs.com/p69lej6ws

**Comment by Razzeee** [comment]
Did this really show up twice? Stop only gets called once here http://www.xbmclogs.com/p69lej6ws#line-13211 and thats absolutely how it should work.

**Comment by RavRob** [comment]
I am testing with other addons.  It seems to work on with Phoenix.  I am
still testing but I'm beginning to think it might be Genesis that is
causing this issue.

On Monday, May 18, 2015, Razzeee <notifications@github.com> wrote:

> Did this really show up twice? Stop only gets called once here
> http://www.xbmclogs.com/p69lej6ws#line-13211 and thats absolutely how it
> should work.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Razzeee/script.trakt/issues/169#issuecomment-103026318>
> .
>


-- 
Sent from Gmail Mobile

**Comment by Razzeee** [comment]
@RavRob 
I have read from another user in the vip forum that he has double scrobbles. Also using Genesis, can you please check this first?

**Comment by RavRob** [comment]
I can confirm now that it is a Genesis problem.  Local content only
scrobbles once as does other addons such as Phoenix or SALTS.

On Tuesday, May 19, 2015, Razzeee <notifications@github.com> wrote:

> @RavRob <https://github.com/RavRob>
> I have read from another user in the vip forum that he has double
> scrobbles. Also using Genesis, can you please check this first?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Razzeee/script.trakt/issues/169#issuecomment-103371598>
> .
>


-- 
Sent from Gmail Mobile

**Comment by lambda81** [comment]
I have it fixed, Genesis also supports internal scrobbling, when Trakt addon added PIN support, Genesis checking of when it should use internal scrobbling or rely on Trakt addon broke.
I have it fixed and the update will be pushed today.

**Comment by RavRob** [comment]
Thanks for this

On Tuesday, May 26, 2015, lambda81 <notifications@github.com> wrote:

> I have it fixed, Genesis also supports internal scrobbling, when Trakt
> addon added PIN support, Genesis checking of when it should use internal
> scrobbling or rely on Trakt addon broke.
> I have it fixed and the update will be pushed today.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Razzeee/script.trakt/issues/169#issuecomment-105477102>
> .
>


-- 
Sent from Gmail Mobile

**Comment by insurw** [comment]
@Razzeee Did the last update fix the duplicates when switching profiles, if not why close this?

**Comment by lambda81** [comment]
it's not released yet, but the problem is fixed, so there is no point razzee having it open. You can wait for some hours, I am working on updates in 3 addons.

**Comment by insurw** [comment]
@lambda81 Huh? Duplicates when using multiple profiles isn't about Genesis. This issue was originally about profiles.

**Comment by lambda81** [comment]
sorry insurw, i didn't see that this is an issue with 2 different issues that lead to the same result.
maybe razzee should open it again.

**Comment by Razzeee** [comment]
@insurw well I'm not 100% sure but I just tried with an installation of isengard and it did work.
Might have to do with the PIN auth as we changed about everything about the auth.

Can you try?

**Comment by insurw** [comment]
It was gone for a while but it seems to be back with Kodi 15.1 and latest trakt addon. I was getting 2-3 duplicates regularly again. I turned back on Kodi restart after profile change and I also had to delay trakt addon start (set it at 15 s). Now I never get duplicates. Though I'm guessing very few/nobody have one trakt account for multiple profiles.

I'll try and fix a logfile later this week, now I'm just glad the duplicates are gone.
